app/controllers/ads/token_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class Ads::TokenController < ApplicationController
+  def new
+    @ad = Ad.active.find(params[:ad_id])
+  end
+
+  def create
+    @ad = Ad.active.find(params[:ad_id])
+    if valid_email_provided?
+      send_token(@ad)
+      redirect_to ad_path(@ad), notice: I18n.t("tokens.sent.#{params[:a]}")
+    else
+      flash.now[:alert] = "Email ne odgovara email adresi ostavljenoj uz oglas."
+      render :new
+    end
+  end
+
+  private
+
+  def send_token(ad)
+    ad.update!(token: SecureRandom.hex)
+    AdMailer.send_token(ad, params[:a], build_url(ad, params[:a])).deliver_later
+  end
+
+  def valid_email_provided?
+    params[:email].present? && @ad.email == params[:email].strip
```

Comment:
Maybe strip the ad e-mail as well when comparing? I don't see any stripping when entering it. Also, e-mails are case insensitive so it should probably also `downcase`.

Reply:
Fixed here https://github.com/vlado/earthquake-croatia/pull/110/commits/a7479c3e4d2a42179fccfb4a5c410de066318e8a#diff-c5370a83701cc53b832ee107cd40eaa74512c78cc978112655e654b393f8f3a7R35